app/src/main/java/com/compscieddy/meetinthemiddle/GroupActivity.java

```diff
@@ -384,7 +402,41 @@ public void onClick(View v) {
         Etils.showToast(GroupActivity.this, "Invite Button Clicked");
         break;
 
-    }
+      case R.id.expand_chat_fab:
+
+        Display display = getWindowManager().getDefaultDisplay();
+        Point size = new Point();
+        display.getSize(size);
+        int height = size.y;
+
+        // todo: use Etils.getScreenHeight() instead
+
+        if (!expanded) {
+          mExpandButton.setImageResource(R.drawable.ic_expand_more_black_48dp);
 
+          RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(
+              RelativeLayout.LayoutParams.MATCH_PARENT,
+              (int) (height * 0.75));
+          params.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
+          mBottomSection.setLayoutParams(params);
+
+          Etils.showToast(GroupActivity.this, "Expand chat");
+
+          expanded = !expanded;
+        } else {
+          mExpandButton.setImageResource(R.drawable.ic_expand_less_black_48dp);
+          Resources r = getResources();
+          RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(
+              RelativeLayout.LayoutParams.MATCH_PARENT,
+              Etils.dpToPx(250));
+          params.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
```

Comment:
move this *out* of the if/else statement since you have identical code in the if and else